src/brogue/Monsters.c

```diff
@@ -2381,6 +2381,15 @@ boolean targetEligibleForCombatBuff(creature *caster, creature *target) {
 // Assumes that the conditions in generallyValidBoltTarget have already been satisfied.
 boolean specificallyValidBoltTarget(creature *caster, creature *target, enum boltType theBoltType) {
 
+    if (rogue.patchVersion >= 3
+        && caster->status[STATUS_DISCORDANT]
+        && caster->creatureState == MONSTER_WANDERING
+        && target == &player
+        && !awareOfTarget(caster, target)) {
+
+        return false;
```

Comment:
This seems overly specific - but I see the difficulty in making it more general. The problem is that the bolt-casting code assumes that the caster is aware of all targets. Normally this isn't a huge issue, because typical creatures only cast when hunting, when they are aware by definition. But discordant creatures are hard-coded to cast spells at all times, in monstersTurn.

So, given that, this works to remove the case for the player.. but it leaves the possibility (though I'm not certain) that discordant enemies can still cast at far-away allies. This problem is further exacerbated by the fact there isn't a comprehensive system of awareness between two monsters; awareOfTarget is only ever called with target=player, and makes assumptions to that effect.

Reply:
Yeah I was sort of surprised to learn that all monsters are aware of each other. I think the specificity of the solution fits  because there is a stealth mechanic for the player which is not currently honored for ranged discordant attackers.

For monsters with a ranged attack, as long the monster has line-of-sight with the target, it is a valid target. I don't see a problem with this because it also works to the player's advantage with ranged allies. They can shoot enemies across the map. Given that, I don't have an issue with discordant monsters likewise targeting allies from across the map.

However, if there is an appetite to buff ally builds generally, perhaps the ally targeting is something that could be separately considered in the mix of options to boost that type of build.

Reply:
Ok, then I think it makes sense to make this change - I think the inconsistency is better than the high yet unintuitive risk of the player being targeted.

I think awareOfTarget is superfluous given that it's highly related to creatureState?

Reply:
It appears that including awareOfTarget would actually be harmful. It's only usage is in updateMonsterState which happens at the beginning of the monster's turn, before the discordant monster logic. Also, it's non-deterministic since there is a 3% chance of a hunting monster losing track of the player. So there would be a chance for a wandering monster to still attack the player if the awareness check succeeded. I removed it accordingly.